web/settings/clients.go

```diff
@@ -0,0 +1,50 @@
+package settings
+
+import (
+	"errors"
+	"net/http"
+
+	"github.com/cozy/cozy-stack/pkg/consts"
+	"github.com/cozy/cozy-stack/pkg/oauth"
+	"github.com/cozy/cozy-stack/web/jsonapi"
+	"github.com/cozy/cozy-stack/web/middlewares"
+	"github.com/cozy/cozy-stack/web/permissions"
+	"github.com/labstack/echo"
+)
+
+func listClients(c echo.Context) error {
+	instance := middlewares.GetInstance(c)
+
+	if err := permissions.AllowWholeType(c, permissions.GET, consts.OAuthClients); err != nil {
+		return err
+	}
+
+	clients, err := oauth.GetAll(instance)
+	if err != nil {
+		return err
+	}
+
+	objs := make([]jsonapi.Object, len(clients))
+	for i, d := range clients {
+		objs[i] = jsonapi.Object(d)
+	}
+	return jsonapi.DataList(c, http.StatusOK, objs, nil)
+}
+
+func revokeClient(c echo.Context) error {
+	instance := middlewares.GetInstance(c)
+
+	if err := permissions.AllowWholeType(c, permissions.DELETE, consts.OAuthClients); err != nil {
```

Comment:
Making client a `Validable Validate(){ return false }` and checking permissions with `permissions.Allow` after getting the client would allow us to give each client a scope to revoke itself, which would be useful in the future.